app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  def update_user_post_counter
+    author.post_counter += 1
+    author.save
+  end
```

Comment:
- [ ] [Optional] It would be better to use `increment!` as increment! method doesn't want us to save it manually. why it is better you can see [here](https://stackoverflow.com/questions/11499110/increment-vs-1). You can change in other places as well.